cloudify_aws/common/utils.py

```diff
@@ -457,11 +456,29 @@ def get_tags_list(node_prop, runtime_prop, input_prop):
     return tags_list
 
 
+def check_region_name(region):
+    region_matcher = re.compile(constants.REGION_REGEX)
+    if not region_matcher.match(region):
+        raise NonRecoverableError(
+            'Looks as wrong region name:{}'
+            .format(repr(region)))
+
+
+def check_availability_zone(zone):
+    zone_matcher = re.compile(constants.AVAILABILITY_ZONE_REGEX)
+    if not zone_matcher.match(zone):
+        raise NonRecoverableError(
+            'Looks as wrong availability zone: {}'
+            .format(repr(zone)))
```

Comment:
1. Let's change the message to "The `AvailabilityZone` provided in `resource_config` does not match the expected availability zone regular expression."
2. Use an index in the format string method, i.e. "{0}".